t/09_qesapdeployment.t

```diff
@@ -789,4 +789,96 @@ subtest '[qesap_prepare_env::qesap_create_aws_config] not solved template and va
     like $contents[0], qr/region = eu-central-1/, "Expected region eu-central-1 is in the config file";
 };
 
+subtest '[qesap_is_job_finished]' => sub {
+    my $qesap = Test::MockModule->new('qesapdeployment', no_auto => 1);
+    my @results = ();
+    $qesap->redefine(script_output => sub {
+            if ($_[0] =~ /100000/) { return "not json"; }
+            if ($_[0] =~ /200000/) { return "{\"job\":{\"state\":\"donaldduck\"}}"; }
+            if ($_[0] =~ /300000/) { return "{\"job\":{\"state\":\"running\"}}"; }
+    });
+
+    $qesap->redefine(get_required_var => sub { return ''; });
+    $qesap->redefine(record_info => sub { note(join(' ', 'RECORD_INFO -->', @_)); });
+
+    push @results, qesap_is_job_finished(100000);
+    push @results, qesap_is_job_finished(200000);
+    push @results, qesap_is_job_finished(300000);
+
+    my @expected_results = (0, 1, 0);
```

Comment:
what about not using this variable here and directly
```suggestion
ok($results[0] == 0, "Correct result if the openqa job status response isn't JSON");
```

Reply:
all cases where a state was not explicitly retrieved are considered to be 'running', in order to avoid shutting down a running VM by, say, an API mistake or connection error. I'll check the docstring and amend it if not clarified.